python/pyspark/sql/pandas/group_ops.py

```diff
@@ -573,7 +568,16 @@ def transformWithStateUDF(
                 statefulProcessorApiClient.set_handle_state(StatefulProcessorHandleState.CLOSED)
                 return iter([])
 
-            result = handle_data_with_timers(statefulProcessorApiClient, key, inputRows)
+            if timeMode != "none":
+                batch_timestamp = statefulProcessorApiClient.get_batch_timestamp()
+                watermark_timestamp = statefulProcessorApiClient.get_watermark_timestamp()
+            else:
+                batch_timestamp = -1
+                watermark_timestamp = -1
```

Comment:
Can we abstract this as a separate method and share in both UDFs to reduce redundant code?